doc/specs/#1256 - Tab tearoff/spec.md

```diff
@@ -0,0 +1,319 @@
+
+#### Hierarchy Example between two Terminals (tab tearoff/merge)
+
+In the one instance, we have this process hierarchy. Two instances of Windows Terminal exist. In Terminal A, the user has started a `cmd.exe` and a `pwsh.exe` tab. In the second instance, the user has started just one `cmd.exe` tab.
+
+```
+- wt.exe (Terminal Instance A) 
+  |- conhost.exe (in PTY mode) - Hosted to A
+  |  |- cmd.exe
+  |- conhost.exe (in PTY mode) - Hosted to A
+     |- pwsh.exe <-- I will be dragged out
+
+- wt.exe (Terminal Instance B)
+  |- conhost.exe (in PTY mode) - Hosted to B 
+     |- cmd.exe 
+```
+
+When the `pwsh.exe` tab is torn off from Instance A and is dropped onto Instance B, the process hierarchy doesn't actually change. The connection details, preferences, and session metadata are passed via the IPC management channels, but to an outside observer, nothing has actually changed.
+
+```
+- wt.exe (Terminal Instance A) 
+  |- conhost.exe (in PTY mode) - Hosted to A
+  |  |- cmd.exe
+  |- conhost.exe (in PTY mode) - Hosted to B
+     |- pwsh.exe <-- I am hosted in B but I'm parented to A
+
+- wt.exe (Terminal Instance B)
+  |- conhost.exe (in PTY mode) - Hosted to B 
+     |- cmd.exe 
+```
+
+I don't believe there are provisions in the Windows OS to reparent applications to a different process.
+
+Additionally, this becomes more interesting when Terminal Instance A dies and B is still running:
+
+```
+- conhost.exe (in PTY mode) - Hosted to B
+  |- pwsh.exe <-- I am hosted in B but I'm parented to A
+
+- wt.exe (Terminal Instance B)
+  |- conhost.exe (in PTY mode) - Hosted to B 
+     |- cmd.exe 
+```
+
+When instance A dies, the `conhost.exe` that was reparented keeps running and now just appears orphaned within the process hierarchy, reporting to the top level under utilities like Process Explorer.
+
+I believe the action plan here would be to implement what we can, observe the state of the world, and correct going forward. We don't have a solid understanding of how many client applications might be impacted by this apparent change. It also might be perfectly OK because the client applications will always remain parented to the same `conhost.exe` even if those `conhost.exe`s don't report up to the correct `wt.exe`.
+
+It is also unclear whether someone might want to write a utility from the outside to discover this hierarchy. I would be inclined to not provide a way to do this without a strong case otherwise because attempting to understand the local machine process hierarchy is a great way to box yourself in when attempting to expand later to encompass remote connections.
+

```

Comment:
Here's a crazy idea: If there's a manager for both the `wt` instances, they could both be children of the manager. When wt_A creates a new tab, it actually asks the _manager_ to CreateProcess, and the manager hands the connection back to wt_A. Then all the conhosts are children of the manager, not a particular `wt` instance. When you drag and drop to wt_B, the conhost doesn't need to get re-parented or anything, only the TermControl does.

Reply:
If the manager does, do all Terminals go down? Or do those conhosts get orphaned/retrieved/reshuffled into a new one?